python/python/lance/tf/data.py

```diff
@@ -265,5 +266,113 @@ def lance_fragments(dataset: Union[str, Path, LanceDataset]) -> tf.data.Dataset:
     )
 
 
+def _ith_batch(i: int, batch_size: int, total_size: int) -> Tuple[int, int]:
+    start = i * batch_size
+    end = tf.math.minimum(start + batch_size, total_size)
+    return (start, end)
+
+
+def lance_batches(
+    dataset: Union[str, Path, LanceDataset],
+    *,
+    shuffle: bool = False,
+    seed: Optional[int] = None,
+    batch_size: int = 1024,
+    skip: int = 0,
+) -> tf.data.Dataset:
+    """
+    Create a ``tf.data.Dataset`` of batch indices for a Lance dataset.
+
+    Parameters
+    ----------
+    dataset : Union[str, Path, LanceDataset]
+        A Lance Dataset or dataset URI/path.
+    shuffle : bool, optional
+        Shuffle the batches, by default False
+    seed : Optional[int], optional
+        Random seed for shuffling, by default None
+    batch_size : int, optional
+        Batch size, by default 1024
+    skip : int, optional
+        Number of batches to skip.
+
+    Returns
+    -------
+    tf.data.Dataset
+        A tensorflow dataset of batch slice ranges. These can be passed to
+        :func:`lance_take_batches` to create a Tensorflow dataset of batches.
+    """
+    if not isinstance(dataset, LanceDataset):
+        dataset = lance.dataset(dataset)
+    num_rows = dataset.count_rows()
+    num_batches = (num_rows + batch_size - 1) // batch_size
+    indices = tf.data.Dataset.range(num_batches, dtype=tf.int64)
```

Comment:
say a 1B dataset, this one can be 1-million batches, so a few MB batch ids?

Reply:
Yeah the user sets the batch size, so they control the memory use here.